### DISCUSSION TITLE
git Data Source failing to sync

### DISCUSSION THREAD

**Discussion by BriacD** [netbox-community/netbox#14371] [Help Wanted!] [answered]
I have set up both a public and private repo to test. Status go to failed after I refresh the page, almost immediately.

I've set up logging via the documentation but nothing is being generated. I'm having trouble trying to find a path forward here. Either my logging setup is wrong/not working. Or this sort of error won't log.

Any tips on how to get more information towards syncing a git repo or finding logs within Netbox would be appreciated.

**Comment by abhi1693** [comment, marked as answer]
Have you installed the dulwich package?

**Reply by BriacD** [reply]
I have not. I didn't see this referenced in the Netbox docs pertaining to data sources but was able to find it after searching. Going to give this a shot and report back.
Thanks for the tip.

**Reply by BriacD** [reply]
I've installed the packaged via this documentation: https://www.dulwich.io/docs/
I've also updated my requirements files via this command: sudo sh -c "echo 'dulwich' >> /opt/netbox/local_requirements.txt"

I've also rebooted Netbox in case it needed it.

I'm still unable to get the sync to work. Do you know of any specific documentation I can reference for this process? I saw that these packages were required in 3.5 prior but option in 3.6 forward.

I am on 3.6

**Reply by ross-cello** [reply]
You'll need to install in the NetBox venv I believe, if you haven't.
```
source /opt/netbox/venv/bin/activate
pip install dulwich
```
You should be able to confirm with `pip list` inside the NB venv. Perhaps do that first and then do the install if not present.

**Reply by candlerb** [reply]
The preferred way is to add dulwich to `/opt/netbox/local_requirements.txt` and then run `./upgrade.sh`. Doing it that way ensures it's installed in future upgrades too.
